### PR TITLE
copr: derive Copy for Decimal

### PR DESCRIPTION
**Pull request by Fullstop000**
Signed-off-by: Fullstop000 <fullstop1005@gmail.com>

###  What have you changed?

Add `#[derive(Copy)]` for `Decimal`

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

**Comment by lonng** [comment]
Is it proper to derive `Copy` for decimal? @breeswish The `Decimal` almost 40 bytes.

**Comment by breeswish** [comment]
@lonng Yes. That's why only `clone()` is subtituted while references are still reserved.

File: components/tidb_query/src/codec/convert.rs [hunk 1]
```diff
@@ -384,7 +384,7 @@ impl ToInt for Decimal {
     #[inline]
     fn to_int(&self, ctx: &mut EvalContext, tp: FieldTypeTp) -> Result<i64> {
         // TODO: avoid this clone
```
Comment (lonng) [inline components/tidb_query/src/codec/convert.rs]: Please remote this comment.

File: components/tidb_query/src/codec/convert.rs [hunk 2]
```diff
@@ -394,7 +394,7 @@ impl ToInt for Decimal {
     #[inline]
     fn to_uint(&self, ctx: &mut EvalContext, tp: FieldTypeTp) -> Result<u64> {
         // TODO: avoid this clone
```
Comment (lonng) [inline components/tidb_query/src/codec/convert.rs]: ditto

**Pull request merged**